app/scripts/metamask-controller.js

```diff
@@ -1857,6 +1859,38 @@ export default class MetamaskController extends EventEmitter {
       }, this.preferencesController.state),
     );
 
+    // MultichainAccountService has subscription for preferences changes
+    this.controllerMessenger.subscribe(
+      'PreferencesController:stateChange',
+      previousValueComparator((prevState, currState) => {
+        const { useExternalServices: prevUseExternalServices } = prevState;
+        const { useExternalServices: currUseExternalServices } = currState;
+        if (prevUseExternalServices !== currUseExternalServices) {
+          // Set basic functionality and trigger alignment when enabled
+          // This single call handles both provider disable/enable and alignment
+          try {
+            this.controllerMessenger
+              .call(
+                'MultichainAccountService:setBasicFunctionality',
+                currUseExternalServices,
+              )
+              .catch((error) => {
+                console.error(
+                  'Failed to set basic functionality on MultichainAccountService:',
+                  error,
+                );
+              });
+          } catch (error) {
+            // Handle case where MultichainAccountService is not available (e.g., in tests)
+            console.warn(
+              'MultichainAccountService:setBasicFunctionality not available:',
+              error.message,
+            );
+          }
```

Comment:
Wait, the outer `try` `catch` looks wrong to me 🤔

We should not guard against this kind of thing, the action should always be available? Unless I missed something?

And if we have a test for this, it's probably invalid too I'd say.